probe-rs/src/architecture/arm/core/exception_handling/armv7m.rs

```diff
@@ -0,0 +1,309 @@
+                };
+                Ok(format!("HardFault handler. Cause: {description}."))
+            }
+            ExceptionReason::MemoryManagementFault => {
+                if let Some(source) =
+                    Cfsr(core.read_word_32(Cfsr::get_mmio_address())?).get_uf_details(core)?
+                {
+                    Ok(source)
+                } else {
+                    Ok("UsageFault handler. Cause: Unknown.".to_string())
+                }
+            }
+            ExceptionReason::BusFault => {
+                if let Some(source) =
+                    Cfsr(core.read_word_32(Cfsr::get_mmio_address())?).get_bf_details(core)?
+                {
+                    Ok(source)
+                } else {
+                    Ok("BusFault handler. Cause: Unknown.".to_string())
+                }
+            }
+            ExceptionReason::UsageFault => {
+                if let Some(source) =
+                    Cfsr(core.read_word_32(Cfsr::get_mmio_address())?).get_uf_details(core)?
+                {
+                    Ok(source)
+                } else {
+                    Ok("MemManage Fault handler. Cause: Unknown.".to_string())
+                }
+            }
+            ExceptionReason::SVCall => Ok("Supervisor call.".to_string()),
+            ExceptionReason::DebugMonitor => Ok("Synchronous Debug monitor fault.".to_string()),
+            ExceptionReason::PendSV => Ok("Pending Supervisor call.".to_string()),
+            ExceptionReason::SysTick => Ok("Systick handler.".to_string()),
+            ExceptionReason::ExternalInterrupt(exti) => Ok(format!("External interrupt #{exti}.")),
+            ExceptionReason::Reserved => {
+                Ok("Reserved by the ISA, and not usable by software.".to_string())
+            }
+        }
+    }
+}
+
+impl<'probe> ExceptionInterface for crate::architecture::arm::core::armv7m::Armv7m<'probe> {
+    fn calling_frame_registers(
+        &mut self,
+        stackframe_registers: &crate::debug::DebugRegisters,
+    ) -> Result<crate::debug::DebugRegisters, crate::Error> {
+        calling_frame_registers(self, stackframe_registers)
+    }
+
+    fn exception_description(
+        &mut self,
+        stackframe_registers: &crate::debug::DebugRegisters,
+    ) -> Result<String, crate::Error> {
+        // Load the provided xPSR register as a bitfield.
+        let exception_number = Xpsr(
+            stackframe_registers
+                .get_register_value_by_role(&crate::core::RegisterRole::ProcessorStatus)?
+                as u32,
+        )
+        .exception_number();
+
+        Ok(format!(
+            "{:?}",
+            ExceptionReason::from(exception_number).expanded_description(self)?
+        ))
+    }
+
+    fn get_exception_info(
+        &mut self,
+        stackframe_registers: &DebugRegisters,
+    ) -> Result<Option<ExceptionInfo>, Error> {
+        cortexm_6_and_7::get_exception_info(self, stackframe_registers)
```

Comment:
Same comment as in the armv6/7 module :)

Reply:
Done :)